posthog/src/main/java/com/posthog/PostHogConfig.kt

```diff
@@ -136,6 +139,9 @@ public open class PostHogConfig(
     @PostHogInternal
     public var snapshotEndpoint: String = "/s/"
 
+    @PostHogInternal
+    public var isClientSDK: Boolean = false
```

Comment:
since this is public config...

I don't actually know what client sdk means in this context...

(does it mean "android" or "java"? in which case is it better to be those strings?)

Reply:
To tack on top - is there a way to create a stateless + stateful core that inherits from stateless for the SDK code here? Similar to what we do in node.

It feels a bit awkward to have this interface where we add stateful bits based on `isClientSDK` in otherwise stateless functions. Makes it way too easy to introduce bugs + hard to reason about what can happen.

Alternatively, maybe we can extract the stateful parts out of the pure java SDK and into the android one? And have a clean interface for java that allows passing in whatever data you want, buuut doesn't depend on knowing whether it's client side or not.

Reply:
I understand the confusion since the JVM runs everywhere.
`client` means any app running on the user's device eg a Desktop app, an Android app, etc.
Sever means, well running on a server, this concept is known in the JVM world I guess unless I made it up years ago and kept it using lol (think of HTTP Server and HTTP Client), also its annotated with `@PostHogInternal` so no one should touch that besides us, with java docs this will be clearer.

I agree with the abstractions not being great, the goal here was a quick win as described in the PR description, not breaking the current interface so we don't have to cut Android v4 and offer a migration plan again. 
The reason is mainly because of [that](https://github.com/PostHog/posthog-android/pull/129#discussion_r1592489047).